packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -156,18 +179,81 @@ class TechnologyController {
 		return response.status(200).send({ success: true });
 	}
 
+	/**
+	 * Delete a technology user.
+	 * DELETE technologies/:idTechnology/users/:idUser
+	 */
+	async deleteTechnologyUser({ params, response }) {
+		const { idTechnology, idUser } = params;
+		let technology;
+		try {
+			technology = await Technology.findOrFail(idTechnology);
+		} catch (error) {
+			throw new ResourceNotFoundException('technology', 400, 'E_RESOURCE_NOT_FOUND');
+		}
+		let user;
+		try {
+			user = await User.findOrFail(idUser);
+		} catch (error) {
+			throw new ResourceNotFoundException('user', 400, 'E_RESOURCE_NOT_FOUND');
+		}
```

Comment:
Acho que uma query aqui não precisa esperar pela outra, o que acha?
```suggestion
       try { 
       const [ technology, user ] = await Promise.all([ await Technology.findOrFail(idTechnology), await User.findOrFail(idUser) ];
       } catch (error) { 
          //throw the exception 
       }
		
```

Reply:
Verdade, mas nesse caso o `await` não é necessário dentro do Promise.all

```javascript
const [ technology, user ] = await Promise.all([ Technology.findOrFail(idTechnology), User.findOrFail(idUser) ];
```

Reply:
Eu fiz a separação para identificar o recurso quando disparar a exceção, se eu usar essa nova notação quando ocorrer a exceção como saberei qual recurso não encontrou?

Reply:
Correto @nicholasio 👍 

Andei a ver a documentação aqui e vi que o findOrFail da thrown da exception `ModelNotFoundException` não consegue pegar a informação do model dentro do catch ? e.g `error.modelName`

Reply:
Se isso for possível nem precisaremos dessa excessão customizada, pois poderiamos pegar o model name dentro do exception handler.

Reply:
Eu vou ver isso, se for possível eu tiro bastante código daí, até então eu só tinha visto o message.

Reply:
Infelizmente não é possível pegar diretamente o model através de `error.model`, mas eu posso extrai-lo da mensagem, só não sei se meu gerente vai autorizar.
`message = E_MISSING_DATABASE_ROW: Cannot find database row for Technology model`
`const model = error.message.split(':')[1].split(' ')[6];`
Um pequena gambiarra para um bem maior.

Reply:
@nicholasio

Reply:
Se conseguirmos uma regex que funciona bem, por mim vale a pena pra reduzir o código.

Por outro lado, podemos retornar o erro original (além da payload que enviamos atualmente) no caso dev/debug esteja ativo, esse tipo de mensagem de erro vai ser melhor só pra desenvolvimento mesmo....

Reply:
:+1: